### PR TITLE
Add express-socket.io-session types

### PR DESCRIPTION
**Pull request by AylaJK**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/express-socket.io-session/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "strict-export-declare-modifiers": false
```
Comment (sheetalkamat) [inline types/express-socket.io-session/tslint.json]: Since this is a new package, please get rid of any rule disabling from linter

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/express-socket.io-session/tslint.json (hunk 1 above)
Comment (AylaJK) [reply types/express-socket.io-session/tslint.json]: @sheetalkamat Thank you for the feedback. However, I'm unsure how to proceed.

If `"strict-export-declare-modifiers": false` is removed, linting fails regardless of what I try:
- with just `function`, I get `expect  TypeScript@next compile error: A 'declare' modifier is required for a top level declaration in a .d.ts file.`
- with `declare function` I get `strict-export-declare-modifiers  Prefer to explicitly write 'export' for an external module export. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md`
- with `export function` I get `expect  TypeScript@next compile error: An export assignment cannot be used in a module with other exported elements.`

This last error comes directly from the fact that the package this definition is for uses `module.exports = function(...) {...}`, and as such I am using `export =` in the definition.

As I am new to typescript, I decided to look at how other DefinitelyTyped definitions handeled `module.exports = function(...) {...}`. All of the definitions I looked at fell into two cases:
- The package includes `"strict-export-declare-modifiers": false`, such as [@types/express](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/express).
- The package excludes `"strict-export-declare-modifiers": false`, such as [@types/cookie-parser](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/cookie-parser), but fails to lint when I test it. (When I try `npm run lint cookie-parser` it fails solely with `strict-export-declare-modifiers` errors.)

Please let me know if there is a strategy for defining such functions that I have missed.

Thanks,
AylaJK

Comment (sheetalkamat) [reply types/express-socket.io-session/tslint.json]: i dont understand what you are asking here. 
i copy pasted your index.d.ts, test file and tsconfig and used default lint json and there were no linting or test errors?

Comment (AylaJK) [reply types/express-socket.io-session/tslint.json]: In that case I'll push with the default tstlint configuration and see if the TravisCI passes.

Comment (AylaJK) [reply types/express-socket.io-session/tslint.json]: And.... TracisCI passed. I'll have to look into why my local machine is producing different linting/testing results.
Thank you for your help!

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**